spotlight-client/src/contentModels/HistoricalPopulationByCategoryMetric.ts

```diff
@@ -0,0 +1,158 @@
+  });
+
+  return missingMonths.map(({ year, monthIndex }) => ({
+    date: new Date(year, monthIndex),
+    count: 0,
+    category,
+  }));
+}
+
+export default class HistoricalPopulationByCategoryMetric extends Metric<HistoricalPopulationByCategoryRecord> {
+  // UI needs to know this in order to configure proper viewing window
+  dataIncludesCurrentMonth?: boolean;
+
+  constructor(
+    props: BaseMetricConstructorOptions<HistoricalPopulationByCategoryRecord>
+  ) {
+    super(props);
+
+    makeObservable(this, {
+      dataIncludesCurrentMonth: observable,
+      dataSeries: computed,
+    });
+  }
+
+  async fetchAndTransform(): Promise<HistoricalPopulationByCategoryRecord[]> {
+    const transformedData = await super.fetchAndTransform();
+
+    // if the current month is completely missing from data, we will assume it is
+    // actually missing due to reporting lag. But if any record contains it, we will
+    // assume that it should be replaced with an empty record when it is missing
+    const includeCurrentMonth = dataIncludesCurrentMonth(transformedData);
+    runInAction(() => {
+      this.dataIncludesCurrentMonth = includeCurrentMonth;
+    });
+
+    const missingRecords: HistoricalPopulationByCategoryRecord[] = [];
+    const categories = riderCategories;
+
+    categories.forEach(({ identifier }) => {
+      let recordsForCategory;
+      recordsForCategory = transformedData.filter(
+        (record) => record.category === identifier
+      );
+
+      missingRecords.push(
+        ...getMissingMonthsForSeries({
+          records: recordsForCategory,
+          includeCurrentMonth,
+          category: identifier as RiderIdentifier,
+        })
+      );
+    });
+
+    transformedData.push(...missingRecords);
+
+    transformedData.sort((a, b) => ascending(a.date, b.date));
+
+    return transformedData;
+  }
+
+  get records(): HistoricalPopulationByCategoryRecord[] | undefined {
+    let recordsToReturn = this.allRecords;
+    if (!recordsToReturn) return undefined;
+
+    return recordsToReturn;
+  }
+
+  get dataSeries(): DataSeries<HistoricalPopulationByCategoryRecord>[] | null {
+    const { records } = this;
+    if (!records) return null;
+
+    const categories = riderCategories;
+
+    return categories.map(({ identifier, label }, index) => ({
+      label,
+      color: colors.dataViz[index],
+      coordinates: records.filter((record) => record.category === identifier),
+    }));
+  }
+
+  // get unknowns(): UnknownsByDate | undefined {
+  //   const { allRecords } = this;
+
+  //   if (!allRecords) return undefined;
+
+  //   const countsByDate = groups(allRecords, (r) => r.date)
+  //     .map(([date, records]) => ({
+  //       date,
+  //       unknowns: countRiderUnknowns(records, (groupedRecords) =>
+  //         sum(groupedRecords, (r) => r.count)
+  //       ),
+  //     }))
+  //     .filter((item) => hasUnknowns(item.unknowns));
+
+  //   return countsByDate.length ? countsByDate : undefined;
+  // }
```

Comment:
This `unknowns` method

Reply:
@nasaownsky I think you're on the right track in that you probably do need a separate count function because this record type does not include demographic fields like the others. You could just have it accept a `HistoricalPopulationByCategoryRecord` as input or it could be more generic and accept anything that `extends {category: string}` or something like that ... it can be a little simpler than the `countUnknowns` function in that it only has to check for unknowns in one field, the `category` field, and the return type could probably just be a number? (you could probably give it a more generic name also since there is nothing Rider-specific about this logic.)

Essentially a version of [this one-line function](https://github.com/Recidiviz/public-dashboard/blob/nasaownsky/518-riders-narrative/spotlight-client/src/contentModels/unknowns.ts#L29) is all you really need here ... if you wanted to go a step further you could factor that out into something that will generically check any field for unknowns, but I think it's fine to just copy and update it also since it's so simple.

Does that help? Let me know if there are any more specific questions I can answer to assist with this.